docs/usage/shoot_high_availability_best_practices.md

```diff
@@ -0,0 +1,431 @@
+
+Anyway, PTSCs are critical to have, but not perfect, so we saw (unsurprisingly, because that's how the scheduler works), that the scheduler may block the deployment of new pods because it takes the decision pod-by-pod (we opened among others [#109364](https://github.com/kubernetes/kubernetes/issues/109364)).
+
+## Pod Affinities and Anti-Affinities
+
+As said, you can combine PTSCs with [pod affinities and/or anti-affinities](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity). Especially [inter-pod (anti-)affinities](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity) may be helpful to place pods *apart*, e.g. because they are fall-backs for each other or you do not want multiple potentially resource-hungry ["best-effort"](https://kubernetes.io/docs/concepts/workloads/pods/pod-qos/#besteffort) or ["burstable"](https://kubernetes.io/docs/concepts/workloads/pods/pod-qos/#burstable) pods side-by-side (noisy neighbor problem), or *together*, e.g. because they form a unit and you want to reduce the failure domain, reduce the network latency, and reduce the costs.
+
+## Topology Aware Hints
+
+While [topology aware hints](https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints) are not directly related to HA, they are very relevant in the HA context. Spreading your workload across multiple zones may increase network latency and cost significantly, if the traffic is not shaped. Topology aware hints (beta since Kubernetes `v1.23`, replacing the now deprecated topology aware traffic routing with topology keys) help to route the traffic within the originating zone, if possible. Basically, they tell `kube-proxy` how to setup your routing information, so that clients can talk to endpoints that are located within the same zone.
+
+Be aware however, that there are some limitations that we discussed with the Kubernetes networking SIG and [Tim Hockin](https://www.linkedin.com/in/tim-hockin-6501072) specifically and that the maintainers are reluctant to drop. Those are called [safeguards](https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints/#safeguards) and if they strike, the hints are off and traffic is routed again randomly. Especially controversial is the balancing limitation as there is the assumption, that the load that hits an endpoint is determined by the allocatable CPUs in that topology zone, but that's not always if even mostly the case (we opened among others [#110714](https://github.com/kubernetes/kubernetes/issues/110714)). So, this limitation hits far too often and your hints are off, but then again, it's about network latency and cost optimization first, so it's better than nothing.
```

Comment:
I would vote to link https://github.com/kubernetes/kubernetes/issues/113731 instead of https://github.com/kubernetes/kubernetes/issues/110714. It should better summarize what are the issues we hit with TopologyAwareHints.

Reply:
You are right, I will add yours on position 1. Thank you!